engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/compile/CustomNodeValidationSpec.scala

```diff
@@ -257,6 +258,28 @@ class CustomNodeValidationSpec extends FunSuite with Matchers with OptionValues
     }
   }
 
+  test("validate nodes after union if validation of part before fails") {
+    val process =  EspProcess(MetaData("proc1", StreamMetaData()), ExceptionHandlerRef(List()), NonEmptyList.of(
+        GraphBuilder
+          .source("sourceId1", "mySource")
+          .filter("invalidFilter", "not.a.valid.expression")
+          .branchEnd("branch1", "join1"),
+        GraphBuilder
+          .branch("join1", "unionTransformer", Some("outPutVar"),
+            List(
+              "branch1" -> List("key" -> "'key1'", "value" -> "#input")
+            )
+          )
+          .processorEnd("stringService", "stringService" , "stringParam" -> "''")
+      ))
+    val validationResult = validator.validate(process)
+
+    validationResult.variablesInNodes("stringService")("outPutVar") shouldBe TypedObjectTypingResult(Map("branch1" -> Typed[String]))
```

Comment:
Can you add two branches with different value's types?

Reply:
added assertion in other test, this one is for different case